src/main/java/dinkplugin/notifiers/LevelNotifier.java

```diff
@@ -84,18 +89,47 @@ private void handleLevelUp(String skill, int level, int xp) {
         int virtualLevel = level < 99 ? level : Experience.getLevelForXp(xp); // avoid log(n) query when not needed
         Integer previousLevel = currentLevels.put(skill, virtualLevel);
 
-        if (!config.notifyLevel() || previousLevel == null || virtualLevel == previousLevel) {
+        if (previousLevel != null && virtualLevel < previousLevel) {
+            // base skill level should never regress; reset notifier state
+            reset();
             return;
         }
 
-        if (virtualLevel < previousLevel) {
-            // base skill level should never regress; reset notifier state
-            reset();
+        // Check normal skill level up
+        checkLevelUp(config.notifyLevel(), skill, previousLevel, virtualLevel);
+
+        // Skip combat level checking if no level up has occurred
+        if (previousLevel == null || virtualLevel <= previousLevel) {
+            if (currentLevels.containsKey(COMBAT_NAME))
+                return;
+        }
+
+        // Check for combat level increase
+        if (COMBAT_COMPONENTS.contains(skill)) {
+            int combatLevel = getCombatLevel();
+            Integer previousCombatLevel = currentLevels.put(COMBAT_NAME, combatLevel);
+            checkLevelUp(config.levelNotifyCombat(), COMBAT_NAME, previousCombatLevel, combatLevel);
+        }
+    }
+
+    private void checkLevelUp(boolean configEnabled, String skill, Integer previousLevel, int currentLevel) {
+        if (previousLevel == null || currentLevel <= previousLevel) {
+            log.trace("Ignoring non-level-up for {}: {}", skill, currentLevel);
+            return;
+        }
+
+        if (!configEnabled) {
+            log.trace("Ignoring level up of {} to {} due to disabled config setting", skill, currentLevel);
+            return;
+        }
+
+        if (!checkLevelInterval(previousLevel, currentLevel)) {
```

Comment:
Should the virtual level check be skipped for combat level since 99 in all skills naturally equals 126 combat level?